test/integration/webhook/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -0,0 +1,178 @@
+			By("Delete Extension")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, extension))).To(Succeed())
+		})
+
+		osc = &extensionsv1alpha1.OperatingSystemConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-osc",
+				Namespace: testNamespace.Name,
+			},
+			Spec: extensionsv1alpha1.OperatingSystemConfigSpec{
+				Purpose: extensionsv1alpha1.OperatingSystemConfigPurposeReconcile,
+				Units: []extensionsv1alpha1.Unit{
+					{
+						Name: "foo",
+						DropIns: []extensionsv1alpha1.DropIn{
+							{
+								Name:    "drop1",
+								Content: "data1",
+							},
+						},
+						FilePaths: []string{"/foo-bar-file"},
+					},
+				},
+				Files: []extensionsv1alpha1.File{
+					{
+						Path: "foo/bar",
+						Content: extensionsv1alpha1.FileContent{
+							Inline: &extensionsv1alpha1.FileContentInline{
+								Encoding: "b64",
+								Data:     "some-data",
+							},
+						},
+					},
+					{
+						Path: "/foo-bar-file",
+						Content: extensionsv1alpha1.FileContent{
+							ImageRef: &extensionsv1alpha1.FileContentImageRef{
+								Image:           "foo-image:bar-tag",
+								FilePathInImage: "/foo-bar-file",
+							},
+						},
+					},
+				},
+			},
+		}
+		configData.ProjectName = testNamespace.Name
+		By("Create OSC")
+		// Eventually is used because webhook may still not be ready to receive traffic
+		Eventually(func() error {
+			return testClient.Create(ctx, osc)
+		}).Should(Succeed())
+		DeferCleanup(func() {
+			By("Delete OSC")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, osc))).To(Succeed())
+		})
+
+	})
+
+	Context("Default Behaviour of extension with a default configuration", func() {
+		BeforeEach(func() {
+			config = &rsyslog.RsyslogRelpConfig{
+				Target: "localhost",
+				Port:   1520,
+			}
+			configData.Target = "localhost"
+			configData.Port = 1520
+		})
+
+		It("should have all expected files attached", func() {
+			expectedFiles := []extensionsv1alpha1.File{
+				{
+					Path: "foo/bar",
+					Content: extensionsv1alpha1.FileContent{
+						Inline: &extensionsv1alpha1.FileContentInline{
+							Encoding: "b64",
+							Data:     "some-data",
+						},
+					},
+				},
+				{
+					Path: "/foo-bar-file",
+					Content: extensionsv1alpha1.FileContent{
+						ImageRef: &extensionsv1alpha1.FileContentImageRef{
+							Image:           "foo-image:bar-tag",
+							FilePathInImage: "/foo-bar-file",
+						},
+					},
+				},
+			}
+			tmpl, err := template.New("config").Parse(string(testdata.GetSimpleRsyslogConfig()))
+			Expect(err).NotTo(HaveOccurred())
+			var simpleRsyslogConfig bytes.Buffer
+			Expect(tmpl.Execute(&simpleRsyslogConfig, configData)).To(Succeed())
+
+			expectedFiles = append(expectedFiles, testdata.GetRsyslogFiles(simpleRsyslogConfig.Bytes(), true)...)
+			expectedFiles = append(expectedFiles, testdata.GetAuditRulesFiles(true)...)
+			Expect(expectedFiles).To(HaveLen(len(osc.Spec.Files)))
+			Expect(expectedFiles).To(Equal(osc.Spec.Files))
```

Comment:
The expected values should aways come last
```suggestion
                        Expect(osc.Spec.Files).To(HaveLen(len(expectedFiles)))
			Expect(osc.Spec.Files).To(Equal(expectedFiles))
```